target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-whw03v2.dts

```diff
@@ -1,184 +1,41 @@
-		led-boot = &led_blue;
-		led-failsafe = &led_red;
-		led-running = &led_blue;
-		led-upgrade = &led_red;
-	};
-
 	// Default bootargs include rootfstype=ext4 and need to be overriden.
 	chosen {
 		bootargs-append = " root=/dev/ubiblock0_0 rootfstype=squashfs ro";
 	};
-
-	soc {
-		ess-tcsr@1953000 {
-			compatible = "qcom,tcsr";
-			reg = <0x1953000 0x1000>;
-			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
-		};
-
-
-		tcsr@1949000 {
-			compatible = "qcom,tcsr";
-			reg = <0x1949000 0x100>;
-			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
-		};
-
-		tcsr@194b000 {
-			compatible = "qcom,tcsr";
-			reg = <0x194b000 0x100>;
-			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
-		};
-
-		tcsr@1957000 {
-			compatible = "qcom,tcsr";
-			reg = <0x1957000 0x100>;
-			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
-		};
-	};
-
-
-	keys {
-		compatible = "gpio-keys";
-
-		reset {
-			label = "reset";
-			gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_RESTART>;
-		};
-	};
 };
 
 
 &tlmm {
-	mdio_pins: mdio-pinmux {
-		mux-1 {
-			pins = "gpio6";
-			function = "mdio";
-			bias-pull-up;
-		};
-
-		mux-2 {
-			pins = "gpio7";
-			function = "mdc";
-			bias-pull-up;
-		};
-	};
-
 	i2c_0_pins: i2c-0-pinmux {
 		pins = "gpio20", "gpio21";
 		function = "blsp_i2c0";
 		bias-disable;
 	};
+};
 
-	serial_0_pins: serial0-pinmux {
-		pins = "gpio16", "gpio17";
-		function = "blsp_uart0";
-		bias-disable;
-	};
-
-	serial_1_pins: serial1-pinmux {
-		pins = "gpio8", "gpio9", "gpio10", "gpio11";
-		function = "blsp_uart1";
-		bias-disable;
-	};
-
-	spi_0_pins: spi-0-pinmux {
-		mux {
-			pins = "gpio13", "gpio14", "gpio15";
-			function = "blsp_spi0";
-			drive-strength = <12>;
-			bias-disable;
-		};
-
-		mux-cs {
-			pins = "gpio12";
-			function = "gpio";
-			drive-strength = <2>;
-			bias-disable;
-			output-high;
-		};
-	};
-
-	spi_1_pins: spi-1-pinmux {
-		mux-1 {
-			pins = "gpio44", "gpio46", "gpio47";
-			function = "blsp_spi1";
-			bias-disable;
-		};
-
-		mux-2 {
-			pins = "gpio31", "gpio45", "gpio49";
-			function = "gpio";
-			bias-pull-up;
-			output-high;
-		};
-
-		host-interrupt {
-			pins = "gpio42";
-			function = "gpio";
-			input;
-		};
-	};
-
-	wifi_0_pins: wifi0-pinmux {
-		pins = "gpio52";
+&spi_1_pins {
+	mux-wake {
+		pins = "gpio31";
 		function = "gpio";
-		drive-strength = <6>;
 		bias-pull-up;
 		output-high;
 	};
-
-	zigbee-0 {
-		gpio-hog;
-		gpios = <29 GPIO_ACTIVE_HIGH>;
-		bias-disable;
-		output-low;
-	};
-
-	zigbee-1 {
-		gpio-hog;
-		gpios = <50 GPIO_ACTIVE_HIGH>;
-		bias-disable;
-		input;
-	};
-
-	bluetooth-enable {
-		gpio-hog;
-		gpios = <32 GPIO_ACTIVE_HIGH>;
-		output-high;
-	};
 };
 
 &mdio {
 	status = "okay";
 	pinctrl-0 = <&mdio_pins>;
 	pinctrl-names = "default";
-	phy-reset-gpios = <&tlmm 19 GPIO_ACTIVE_LOW>;
-};
-
-&ethphy0 {
-	status = "disabled";
-};
-
-&ethphy1 {
-	status = "disabled";
-};
 
-&ethphy2 {
-	status = "disabled";
+	phy-reset-gpios = <&tlmm 19 GPIO_ACTIVE_LOW>;
```

Comment:
This property does not exist

Reply:
Only reset-gpios is valid.

Reply:
@robimarko thanks! well this line currently [exists in v2](https://github.com/openwrt/openwrt/blob/ecb3859cc92d2e670847934c74255a4453915beb/target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-whw03v2.dts#L178), just like the above sections enabling usb silicon that is not implemented in the hardware.

these are bugs in V2. i don't have a V2 right now so i can't debug it. i'm trying to merge V1 here.

Reply:
@robimarko come to think about it, removing `stdout-path` in the second commit was probably a mistake. IMHO it would be better not to introduce any intentional changes to V2 with the merge. i would like to undo that change if you agree.

i think V2 changes should be left up to someone that can test them. i can create an issue documenting the required V2 changes after we merge this change set, in the hope that somebody picks it up.

Reply:
@robimarko 

i reworked the commits slightly:

1) looking at the stock DTBs of V1 and V2, i realized spi0 was not being set up correctly, and that there was an unsupported flash lurking in V2 (but not in V1). i corrected this for V1 (but in V1, spi0 is disabled anyway).
2) given that i can't test V2, i undid the small non-stylistic changes i had done to it, as i don't want to be responsible for breaking the port. now V2's DTB should be indistinguishable from before the merge (or at least that was my intention).
3) and for documentation purposes, i added a [commit](https://github.com/Lanchon/openwrt/commit/6db0a352b1a248a9c5e3b83f21d2751abe93b469) (not being PR'ed) with the changes i think should be applied to V2. (but again, i will not PR those changes.)

i think the 3 commits of this PR should be ready for merging into main (separately). i will cherrypick only the first one into 23.05 later.

thanks!

Reply:
FYI...

i was worried that maybe i had made some mistake in merging the dts'es.

so for each device i took its merged dts and the dtsi, worked back exclusively from them (but guided by a diff view to the original dts), and i eventually reconstructed the original dts down to the letter. so now i am very confident that the merger is correct.